internal/controllers/addon/utils_test.go

```diff
@@ -128,3 +129,286 @@ func (m *csvEventHandlerMock) ReplaceMap(
 	args := m.Called(addon, csvKeys)
 	return args.Bool(0)
 }
+
+func TestParseAddonInstallConfigurationForAdditionalCatalogSources(t *testing.T) {
+	testCases := []struct {
+		addon                   *addonsv1alpha1.Addon
+		additionalCatalogSource []addonsv1alpha1.AdditionalCatalogSource
+		targetNamespace         string
+		pullSecretName          string
+		stop                    bool
```

Comment:
Nice work! Looks good, just a small comment :)

WDYT about moving these fields into a separate `expected` sub-object? Just for the sake of better readability / understanding. Example:
```go
	testCases := []struct {
		addon    *addonsv1alpha1.Addon
		expected struct {
			additionalCatalogSource []addonsv1alpha1.AdditionalCatalogSource
			targetNamespace         string
			pullSecretName          string
			stop                    bool
		}
	}{...}
```

Reply:
Thanks for the suggestion. I am working on it :)